tests/test_utils_misc/__init__.py

```diff
@@ -95,14 +96,44 @@ class TestItem(Item):
             list(arg_to_iter(TestItem(name="john"))), [TestItem(name="john")]
         )
 
-    def test_create_instance(self):
-        settings = mock.MagicMock()
+    def test_build_from_crawler(self):
         crawler = mock.MagicMock(spec_set=["settings"])
         args = (True, 100.0)
         kwargs = {"key": "val"}
 
+        def _test_with_crawler(m, crawler):
+            build_from_crawler(m, crawler, *args, **kwargs)
+            if hasattr(m, "from_crawler"):
+                m.from_crawler.assert_called_once_with(crawler, *args, **kwargs)
+                self.assertEqual(m.call_count, 0)
+            else:
```

Comment:
So this test only checks `from_crawler` while `test_build_from_settings` checks `from_crawler` and `from_settings`, I think this is an error, this one should check both while `test_build_from_settings` should only check `from_settings`.